department-of-veterans-affairs/caseflow
# Ensure case search returns legacy appeals for vso employees

Resolves #11650

### Description
Adds legacy appeals to the appeals returned from `veteran.accessible_appeals_for_poa`

|Before|After|
|---|---|
|<img width="729" alt="Screen Shot 2019-08-15 at 2 37 48 PM" src="https://user-images.githubusercontent.com/45575454/63117795-7e6caa00-bf6a-11e9-93e8-0e01b7ac5d48.png">|<img width="760" alt="Screen Shot 2019-08-15 at 2 38 13 PM" src="https://user-images.githubusercontent.com/45575454/63117772-76ad0580-bf6a-11e9-9019-f12d5bce25ea.png">|




### Acceptance Criteria
- [ ] The search functionality for VSO users should return results from `LegacyAppeals`, in addition to the `Appeals` that it currently returns.

### Testing Plan
1. Log is as our good friend BILLIE_VSO
2. Find a veteran Billie would not normally have access to
```ruby
> u = User.find_by(css_id: "BILLIE_VSO")
> v = Veteran.find(46)
> CaseSearchResultsForVeteranFileNumber.new(file_number_or_ssn: v.file_number, user: u).call.extra[:search_results][:appeals].any?
=> false
```
3. Pretend Billie's VSO is the veteran's poa by changing [this participant_id](https://github.com/department-of-veterans-affairs/caseflow/blob/a47e65c1cf97a722b24d075c02d66416f8cb7bec/lib/fakes/bgs_service.rb#L554) to Billie's VSO's participant_id. (rails console and server may need to be restarted after this step)
```ruby
> u.organizations.first.participant_id
=> "2452415"
```
4. Ensure this appeal is now returned in the search
```ruby
CaseSearchResultsForVeteranFileNumber.new(file_number_or_ssn: v.file_number, user: u).call.extra[:search_results][:appeals].any?
=> true
```
5. Visit /search and ensure this case shows up when searching for the veteran's file number
```ruby
> v.file_number
=> "228081153"
```
